src/shim/actors/verifreg/state.rs

```diff
@@ -249,109 +153,55 @@ impl VerifiedRegistryStateExt for State {
-                    claims.for_each(actor_id, |k, v| {
-                        let claim_id = fil_actors_shared::v11::parse_uint_key(k)?;
-                        result.insert(claim_id, v.into());
-                        Ok(())
-                    })?;
-                }
-            }
-            Self::V12(s) => {
-                let mut claims = s.load_claims(store)?;
-                let mut actors = vec![];
-                claims.for_each(|k, _| {
-                    let actor_id = fil_actors_shared::v12::parse_uint_key(k)?;
-                    actors.push(actor_id);
-                    Ok(())
-                })?;
-
-                for actor_id in actors {
-                    claims.for_each_in(actor_id, |k, v| {
-                        let claim_id = fil_actors_shared::v12::parse_uint_key(k)?;
-                        result.insert(claim_id, v.into());
-                        Ok(())
-                    })?;
-                }
-            }
-            Self::V13(s) => {
-                let mut claims = s.load_claims(store)?;
-                let mut actors = vec![];
-                claims.for_each(|k, _| {
-                    let actor_id = fil_actors_shared::v13::parse_uint_key(k)?;
-                    actors.push(actor_id);
-                    Ok(())
-                })?;
-
-                for actor_id in actors {
-                    claims.for_each_in(actor_id, |k, v| {
-                        let claim_id = fil_actors_shared::v13::parse_uint_key(k)?;
-                        result.insert(claim_id, v.into());
-                        Ok(())
-                    })?;
-                }
-            }
-            Self::V14(s) => {
-                let mut claims = s.load_claims(store)?;
-                let mut actors = vec![];
-                claims.for_each(|k, _| {
-                    let actor_id = fil_actors_shared::v14::parse_uint_key(k)?;
-                    actors.push(actor_id);
-                    Ok(())
-                })?;
-
-                for actor_id in actors {
-                    claims.for_each_in(actor_id, |k, v| {
-                        let claim_id = fil_actors_shared::v14::parse_uint_key(k)?;
-                        result.insert(claim_id, v.into());
-                        Ok(())
-                    })?;
-                }
-            }
+            State::V9(state) => list_all_inner_old!(state, store, v9, load_claims, result),
+            State::V10(state) => list_all_inner_old!(state, store, v10, load_claims, result),
+            State::V11(state) => list_all_inner_old!(state, store, v11, load_claims, result),
+            State::V12(state) => list_all_inner!(state, store, v12, load_claims, result),
+            State::V13(state) => list_all_inner!(state, store, v13, load_claims, result),
+            State::V14(state) => list_all_inner!(state, store, v14, load_claims, result),
         };
         Ok(result)
     }
 }
+
+#[macro_export]
```

Comment:
Do we have to export these macros?

Reply:
If we don't - the macros have to be defined strictly above the code that uses them, which is inconvenient.

Reply:
this exports the macro as a public item at the root of the crate. Does `pub(crate) use list_all_inner` below the macro work? Else I think moving the definition to the top of the file is the right thing to do